guillotina/middlewares/errors.py

```diff
@@ -28,28 +33,42 @@ async def _send(msg):
             resp = await self.next_app(scope, receive, _send)
         except Exception as exc:
             request = task_vars.request.get()
-            view_result = self.generate_error_response(exc, None, "ServiceError")
+            view_result = generate_error_response(exc, bubble=False)
             resp = await apply_rendering(View(None, request), request, view_result)
             if headers_sent:
                 # Too late to send status 500, headers already sent
                 raise
         return resp
 
-    def generate_error_response(self, e, request, error, status=500):
-        # We may need to check the roles of the users to show the real error
-        eid = uuid.uuid4().hex
-        if isinstance(e, asyncio.CancelledError):  # pragma: no cover
-            message = _("Cancelled execution of view") + " " + eid
-            logger.warning(message, exc_info=e, eid=eid, request=request)
-        else:
-            message = _("Error on execution of view") + " " + eid
-            logger.error(message, exc_info=e, eid=eid, request=request)
-        data = {
-            "message": message,
-            "reason": error_reasons.UNKNOWN.name,
-            "details": error_reasons.UNKNOWN.details,
-            "eid": eid,
-        }
-        if app_settings.get("debug"):
-            data["traceback"] = traceback.format_exc()
-        return response.HTTPInternalServerError(content=data)
+
+def generate_error_response(
```

Comment:
@masipcat @vangheem this bubbling logic was removed from 5.x to 6.x and it seems that is needed in guillotina_batch https://github.com/guillotinaweb/guillotina_batch/blob/master/guillotina_batch/api.py#L220

does this make sense?

Reply:
maybe I could just re-write all of this in the guillotina_batch

Reply:
The idea was that `generate_error_response` only take care of unhandled exceptions (or exceptions that aren't a `Response`). I'd say that maybe you want to do something like this in guillotina_batch: https://github.com/plone/guillotina/commit/fd10e07f53536b77685e981d760871fe74fe8c85#diff-aab636c0eb9b37096c447b1e16abbe46R164-R175

Reply:
yes! that makes sense. however, I will need to deal with unhandled exceptions there too instead of in the middleware.

thanks @masipcat